crates/sui-bridge/src/eth_syncer.rs

```diff
@@ -156,16 +156,23 @@ where
                 continue;
             };
             let len = events.len();
-            if !events.is_empty() {
-                // Note: it's extremely critical to make sure the Logs we send via this channel
-                // are complete per block height. Namely, we should never send a partial list
-                // of events for a block. Otherwise, we may end up missing events.
-                events_sender
-                    .send((contract_address, events))
-                    .await
-                    .expect("All Eth event channel receivers are closed");
-                tracing::info!(?contract_address, "Observed {len} new Eth events",);
-            }
+
+            // Note 1: we send an empty list of events to the channel. This is because of how
```

Comment:
Maybe rephrase this to something like: we always send events to the channel even when it's empty..

Reply:
fixed in https://github.com/MystenLabs/sui/pull/15856